rosidl_generator_c/resource/msg__functions.c.em

```diff
@@ -112,13 +120,23 @@ for field in spec.fields:
         if not field.type.is_primitive_type() or field.type.type == 'string':
             # initialize each array element
             lines.append('for (size_t i = 0; i < %d; ++i) {' % field.type.array_size)
-            lines.append('  %s__init(&msg->%s[i]);' % (get_typename_of_base_type(field.type), field.name))
+            lines.append('  if(!%s__init(&msg->%s[i])) {' % (get_typename_of_base_type(field.type), field.name))
+            lines.append('    %s__destroy(msg);' % msg_typename)
+            lines.append('    return false;')
+            lines.append('  }')
             lines.append('}')
 
-    else:
+    else:  # dynamic array
         if field.default_value is None:
             # initialize the dynamic array with a capacity of zero
-            lines.append('%s__Array__init(&msg->%s, 0);' % (get_typename_of_base_type(field.type), field.name))
+            lines.append('if (!%s__Array__init(&msg->%s, 0)) {' % (get_typename_of_base_type(field.type), field.name))
+            lines.append('  %s__destroy(msg);' % array_typename)
+            lines.append('  return false;')
+            lines.append('}')
+            cleanup_line = '%s__Array__destroy(&msg->%s);' % (get_typename_of_base_type(field.type), field.name)
```

Comment:
Not to sel + @allenh1 : this expects an array so generates warnings when we pass it a non Array message structure